README.md

```diff
@@ -346,6 +346,129 @@ end
 
 For an example of advanced integration that supports multiple teams, see [slack-gamebot](https://github.com/dblock/slack-gamebot) and [playplay.io](http://playplay.io) that is built on top of it.
 
+### Model-View-Controller Design
+
+The `command` method is essentially a controller method that receives input from the outside and acts upon it. Complex behaviors could lead to a long and difficult-to-understand `command` block. A complex `command` block is a candidate for separation into classes conforming to the Model-View-Controller pattern popularized by Rails.
+
+The library provides three helpful base classes named `SlackRubyBot::MVC::Model::Base`, `SlackRubyBot::MVC::View::Base`, and `SlackRubyBot::MVC::Controller::Base`.
+
+Testing a `command` block is difficult. As separate classes, the Model/View/Controller's behavior can be tested via `rspec` or a similar tool.
+
+#### Controller
+
+The Controller is the focal point of the bot behavior. Typically the code that would go into the `command` block will now go into an instance method in a Controller subclass. The instance method name should match the command name exactly (case sensitive).
+
+As an example, these two classes are functionally equivalent.
+
+```
+class Agent < SlackRubyBot::Bot
+  command 'sayhello' do |client, data, match|
+    client.say(channel: data.channel, text: "Received command #{match[:command]} with args #{match[:expression]}")
+  end
+end
+
```

Comment:
It's probably clearer to split these two blocks, for example

Consider the following `Agent` command.

# code goes here

With the MVC pattern it inherits from ....

# code goes here.